sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashMapGenerator.scala

```diff
@@ -174,6 +174,7 @@ abstract class HashMapGenerator(
           """
         }
       case StringType => hashBytes(s"$input.getBytes()")
+      case CalendarIntervalType => hashInt(s"$input.hashCode()")
```

Comment:
looking at the code here, I think we need a dedicated hash implementation for calendar interval, to return an int.

Reply:
isn't this already calling the `hashcode` method of the calendar interval?

Reply:
OK this seems fine, the hash result can be either int or long in the generated code.